packages/react-ui-components/src/IconButton/iconButton.tsx

```diff
@@ -0,0 +1,91 @@
+import {IconName} from '@fortawesome/fontawesome-svg-core';
+import mergeClassNames from 'classnames';
+import React, {PureComponent} from 'react';
+
+import {PickDefaultProps} from '../../types';
+import {ButtonHoverStyle, ButtonProps, ButtonSize, ButtonStyle} from '../Button/button';
+import {IconProps} from '../Icon/icon';
+
+interface IconButtonTheme {
+    readonly 'iconButton': string;
+    readonly [key: string]: string;
+}
+
+interface IconButtonProps {
+    /**
+     * The icon key which gets passed to the Icon Component.
+     */
+    readonly icon: string | IconName;
+
+    /**
+     * An optional `className` to attach to the wrapper.
+     */
+    readonly className?: string;
+
+    /**
+     * Defines the size of the icon button.
+     */
+    readonly size: ButtonSize;
+
+    /**
+     * An optional css theme to be injected.
+     */
+    readonly theme?: IconButtonTheme;
+
+    // TODO: This feels strange. Can we use the actual class here?
+    /**
+     * Static component dependencies which are injected from the outside (index.js)
+     */
+    readonly IconComponent: React.ComponentClass<IconProps>;
```

Comment:
This feels strange, but we can not change the current setup where we inject Components as Props as we do right now, right?

Reply:
Well, it's a feature of react themr, a kind of dependency inject. But why does it feel strange?

Reply:
Wait, not themr in fact, but we did it to avoid issues in tests, a dependency injection actually. I think there's a way to get rid of it, but I don't see why we should do that.

Reply:
For me it feels really strange. If it were a function as prop that renders a component, fine with me. But a component class as prop just feels off. Maybe it's just me but I'll have a look at this later.